languages/thingtalk/ast_manip.ts

```diff
@@ -1467,54 +1401,50 @@ function addGetPredicateJoin(table : Ast.Expression,
         return addFilterInternal(table, newAtom, {});
     }
 
-    let newAtom = new Ast.BooleanExpression.Atom(null, 'id', (lhsArg.type.isArray ? 'in_array' : '=='), new Ast.Value.VarRef(lhsArg.name));
-    if (negate)
-        newAtom = new Ast.BooleanExpression.Not(null, newAtom);
-    const get_predicate = new Ast.BooleanExpression.External(null,
-        get_predicate_table.expression.invocation.selector,
-        get_predicate_table.expression.invocation.channel,
-        get_predicate_table.expression.invocation.in_params,
-        new Ast.BooleanExpression.And(null, [get_predicate_table.filter, newAtom]),
-        get_predicate_table.expression.invocation.schema
+    // add id projection to subquery
+    if (subquery instanceof Ast.ProjectionExpression)
+        subquery.args = ['id'];
+    else
+        subquery = new Ast.ProjectionExpression(null, subquery, ['id'], [], [], subquery.schema);
+
+    const comparisonSubquery = new Ast.BooleanExpression.ComparisonSubquery(
+        null,
+        new Ast.Value.VarRef(lhsArg.name),
+        lhsArg.type.isArray ? 'contains' : '==',
```

Comment:
Should this be `contains~` and `=~` if the type is string?

Reply:
yes, I will fix this

Reply:
oh actually this basic version only generate id filter, so this is not an issue for this pr